core/src/main/scala/sttp/tapir/FileRange.scala

```diff
@@ -23,3 +30,7 @@ case class RangeValue(start: Option[Long], end: Option[Long], fileSize: Long) {
     case _                          => None
   }
 }
+trait InputStreamSupplier {
+  def openStream(): InputStream
+}
+case class ResourceRange(inputStreamSupplier: InputStreamSupplier, range: Option[RangeValue] = None) extends StaticResource
```

Comment:
maybe this should simply be `InputStreamRange(inputStream: () => InputStream, range)`? Then we wouldn't have `Resource` anymore (as it's more general than that)

Reply:
Done